server/routes/checkYourAnswers/checkYourAnswersPage.ts

```diff
@@ -0,0 +1,153 @@
+  constructor(private readonly pageType: PageRequestType) {}
+
+  isCreation(): boolean {
+    return this.pageType !== PageRequestType.EDIT_REPORTER && this.pageType !== PageRequestType.EDIT_REVIEWER
+  }
+
+  isEditByReporter(): boolean {
+    return this.pageType === PageRequestType.EDIT_REPORTER
+  }
+
+  isEditByReviewer(): boolean {
+    return this.pageType === PageRequestType.EDIT_REVIEWER
+  }
+}
+
+const getVariablesForPageType = (
+  pageOptions: PageOptions,
+  prisonerNumber: string,
+  adjudicationNumber: number,
+  incidentDetailsData: CheckYourAnswers
+) => {
+  if (pageOptions.isEditByReporter()) {
+    return {
+      editIncidentDetailsURL: `/incident-details/${prisonerNumber}/${adjudicationNumber}/submitted/edit?referrer=/check-your-answers/${prisonerNumber}/${adjudicationNumber}/report`,
+      editIncidentStatementURL: `/incident-statement/${prisonerNumber}/${adjudicationNumber}/submitted/edit`,
+      exitUrl: `/prisoner-report/${prisonerNumber}/${incidentDetailsData.adjudicationNumber}/report`,
+    }
+  }
+  if (pageOptions.isEditByReviewer()) {
+    return {
+      editIncidentDetailsURL: `/incident-details/${prisonerNumber}/${adjudicationNumber}/submitted/edit?referrer=/check-your-answers/${prisonerNumber}/${adjudicationNumber}/review`,
+      exitUrl: `/prisoner-report/${prisonerNumber}/${incidentDetailsData.adjudicationNumber}/review`,
+    }
+  }
+  return {
+    editIncidentDetailsURL: `/incident-details/${prisonerNumber}/${adjudicationNumber}/edit`,
+    editIncidentStatementURL: `/incident-statement/${prisonerNumber}/${adjudicationNumber}`,
+    exitUrl: `/place-the-prisoner-on-report/${prisonerNumber}/${adjudicationNumber}`,
+  }
+}
+
+const getRedirectUrls = (pageOptions: PageOptions, completeAdjudicationNumber: number) => {
+  if (pageOptions.isEditByReporter())
+    return `/prisoner-placed-on-report/${completeAdjudicationNumber}/changes-confirmed/report`
+  if (pageOptions.isEditByReviewer())
+    return `/prisoner-placed-on-report/${completeAdjudicationNumber}/changes-confirmed/review`
+  return `/prisoner-placed-on-report/${completeAdjudicationNumber}`
+}
+
+const getErrorRedirectUrl = (pageOptions: PageOptions, prisonerNumber: string, adjudicationNumber: number) => {
+  if (pageOptions.isEditByReporter()) return `/your-completed-reports`
+  if (pageOptions.isEditByReviewer()) return `/all-completed-reports`
+  return `/place-the-prisoner-on-report/${prisonerNumber}/${adjudicationNumber}`
+}
+
+export default class CheckYourAnswersPage {
+  pageOptions: PageOptions
+
+  constructor(
+    pageType: PageRequestType,
+    private readonly placeOnReportService: PlaceOnReportService,
+    private readonly locationService: LocationService,
+    private readonly decisionTreeService: DecisionTreeService
+  ) {
+    this.pageOptions = new PageOptions(pageType)
+  }
+
+  private renderView = async (req: Request, res: Response, pageData: PageData): Promise<void> => {
+    const { error } = pageData
```

Comment:
Just a suggestion, but I converted the req and res stuff to data structures at the beginning (and the end) of the main methods (view and submit). That way it makes it clear what is expected from the req/res and what we are putting on it.

Having said all that, this is pretty clear in this page, so it's not that important.